src/TimeSpans.jl

```diff
@@ -34,6 +34,9 @@ struct TimeSpan
         start < stop || throw(ArgumentError("start(span) < stop(span) must be true, got $start and $stop"))
         return new(start, stop)
     end
+    function TimeSpan(start::Union{Period,Dates.CompoundPeriod}, stop::Union{Period,Dates.CompoundPeriod})
+        return TimeSpan(Dates.value(convert(Nanosecond, start)), Dates.value(convert(Nanosecond, stop)))
```

Comment:
Since `TimeSpan` stores its endpoints as `Nanosecond`s, you don't need the `Dates.value` calls here since you're already `convert`ing to `Nanosecond`; it will just get re-wrapped by the other method anyway.
```suggestion
        return TimeSpan(convert(Nanosecond, start), convert(Nanosecond, stop))
```

Reply:
Also worth noting that this doesn't need to be an inner constructor but it also doesn't really matter

Reply:
I think we still need it to be, so that it goes through the range checking...

Reply:
Outer constructors can (and (almost?) always?) call inner constructors, so as long as you're returning `TimeSpan(...)` and not `new(...)` (which is only allowed in inner constructors), it will go through the appropriate checks.

Reply:
Ah, gotcha. Will update!

Reply:
updated